src/libosd/cl_cdm.c

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2018 The Open SoC Debug Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <osd/cl_cdm.h>
+
+#include <assert.h>
+#include <osd/osd.h>
+#include <osd/reg.h>
+#include <osd/packet.h>
+#include "osd-private.h"
+
+static struct osd_cdm_event* build_cdm_event(const struct osd_cdm_desc *cdm_desc,
+                                             const struct osd_packet *pkg)
+{
+    struct osd_cdm_event *ev = calloc(1, sizeof(struct osd_cdm_event));
+
+    size_t exp_payload_len = 1; //stall
```

Comment:
style: add a space after the comment: `// stall`.

You can use clang-format to lint your code, a corresponding lint rule is part of the source tree.